synapse/storage/data_stores/main/push_rule.py

```diff
@@ -656,12 +662,31 @@ def set_push_rule_enabled(self, user_id, rule_id, enabled):
                 user_id,
                 rule_id,
                 enabled,
+                is_default_rule,
             )
 
     def _set_push_rule_enabled_txn(
-        self, txn, stream_id, event_stream_ordering, user_id, rule_id, enabled
+        self,
+        txn,
+        stream_id,
+        event_stream_ordering,
+        user_id,
+        rule_id,
+        enabled,
+        is_default_rule,
     ):
         new_id = self._push_rules_enable_id_gen.get_next()
+
+        if not is_default_rule:
+            try:
+                # first check it exists
+                self.db.simple_select_one_onecol_txn(
+                    txn, "push_rules", {"user_name": user_id, "rule_id": rule_id}, "id"
+                )
+            except StoreError as serr:
+                if serr.code == 404:
+                    raise StoreError(404, "Push rule does not exist.", Codes.NOT_FOUND)
```

Comment:
this feels a bit redundant?

(if the objective is to make the error nicer, it might be better to do this at a higher level?)

Reply:
Objective was to make it have a code of `NOT_FOUND` rather than `UNKNOWN`. Comment worthwhile?

Reply:
gah, I see. Well, can you use `NotFoundError`, and document it in the docstring for `set_push_rule_enabled` ?